controller/api/public/server.go

```diff
@@ -73,14 +73,16 @@ func (h *handler) ServeHTTP(w http.ResponseWriter, req *http.Request) {
 
 	// Serve request
 	switch req.URL.Path {
-	case ApiRoot + ApiPrefix + "Stat":
+	case fullUrlPathFor("Stat"):
 		h.handleStat(w, req)
-	case ApiRoot + ApiPrefix + "Version":
+	case fullUrlPathFor("Version"):
 		h.handleVersion(w, req)
-	case ApiRoot + ApiPrefix + "ListPods":
+	case fullUrlPathFor("ListPods"):
 		h.handleListPods(w, req)
-	case ApiRoot + ApiPrefix + "Tap":
+	case fullUrlPathFor("Tap"):
 		h.handleTap(w, req)
+	case fullUrlPathFor("SelfCheck"):
+		h.handleSelfCheck(w, req)
```

Comment:
These are static strings. Maybe it makes sense to define them as vars at the top of this file, rather than rebuild them each time.

```
var (
	jsonMarshaler   = jsonpb.Marshaler{EmitDefaults: true}
	jsonUnmarshaler = jsonpb.Unmarshaler{}
	statPath        = fullUrlPathFor("Stat")
	versionPath     = fullUrlPathFor("Version")
	listPodsPath    = fullUrlPathFor("ListPods")
	tapPath         = fullUrlPathFor("Tap")
	selfCheckPath   = fullUrlPathFor("SelfCheck")
)
```

And then use the vars here.